### DISCUSSION TITLE
Issue installation on WSL

### DISCUSSION THREAD

**Discussion by MariannaDellOtto** [firedrakeproject/firedrake#3588] [Firedrake support] [answered]
Hello,
I am trying to install firedrake on WSL. 
Following the specific instructions (https://github.com/firedrakeproject/firedrake/wiki/Installing-on-Windows-Subsystem-for-Linux), I first updated my Ubuntu installation. Then, since I have Anaconda for Python installed at C:, I unmounted /mnt/c/ before installing Firedrake. 
Then I installed Firedrake:
curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/master/scripts/firedrake-install
python3 firedrake-install

However it encounters some problem installing petsc:

 Installing petsc/
Depending on your platform, PETSc may take an hour or more to build!
Traceback (most recent call last):
  File "/home/marienna/firedrake-install", line 1864, in <module>
    install("petsc/")
  File "/home/marienna/firedrake-install", line 1058, in install
    build_and_install_petsc()
  File "/home/marienna/firedrake-install", line 1170, in build_and_install_petsc
    check_call([python, "./configure", "PETSC_DIR={}".format(petsc_dir), "PETSC_ARCH={}".format(petsc_arch)] + petsc_options)
  File "/home/marienna/firedrake-install", line 674, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
  File "/usr/lib/python3.10/subprocess.py", line 421, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/home/marienna/firedrake/bin/python', './configure', 'PETSC_DIR=/home/marienna/firedrake/src/petsc', 'PETSC_ARCH=default', '--with-shared-libraries=1', '--download-cmake', '--with-c2html=0', '--download-hdf5', '--download-superlu_dist', '--download-scalapack', '--download-suitesparse', '--with-zlib', '--with-fortran-bindings=0', '--download-pastix', '--download-hwloc', '--download-netcdf', '--download-ptscotch', '--download-metis', '--download-mumps', '--download-mpich', '--download-bison', '--download-pnetcdf', '--with-debugging=0', 'hypre']' returned non-zero exit status 1.

When I activate the venv firedrake, looking at the installed packages, I can not find firedrake. Moreover, trying the test, they do not run, with the following error:
ImportError while loading conftest '/home/marienna/firedrake/src/firedrake/tests/conftest.py'.
tests/conftest.py:4: in <module>
    from firedrake.petsc import get_external_packages
E   ModuleNotFoundError: No module named 'firedrake'

Furthermore, in the folder petsc there is no file 'make.log'.

[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/15434519/firedrake-install.log)
[configure.log](https://github.com/firedrakeproject/firedrake/files/15434756/configure.log)

Thank you a lot for your help

**Comment by connorjward** [comment, marked as answer]
Your installation crashed on PETSc install without any sort of clear error message. I wonder if you are running out of memory. Can you try again but this time run `$ export PETSC_CONFIGURE_OPTIONS="--with-make-np=2"` beforehand?

**Reply by MariannaDellOtto** [reply]
It could be a memory problem indeed. 
With the command suggested by you it is installed without any problem. Thank you for the help!
